blog-posts/event-sourcing-serveless/event-sourcing-serveless.md

```diff
@@ -0,0 +1,171 @@
+---
+published: false
+title: 'A prod-proof AWS serverless event-sourcing architecture has never been so simple to implement'
+cover_image:
+description: 'Description of the article'
+tags: serverless, eventsourcing, AWS, CQRS
+series:
+canonical_url:
+---
+
+# Introduction
+
+You probably use a bank application in your every day life and you wonder how they work. Every credit and debit transaction you make is stored as an event in a data store. This event data store is the source of truth and the current state of your account is computed from all events stored in the datastore. This is known as event sourcing.
+
+<img src="./assets/compute-event-sourcing-state.png" style="display: block; margin: auto" width="200"/>
+
+Numerous applications need to store a trace of all events to use the history or even for legal purposes such as a bank. If you share this need and you're thinking about implementing it in serverless, you are in the right place. Thus I want to share with you, 2 years of learnings that will lead you to have the most stable and the less painful solution you can go with.
+
+## Event sourcing's best friend interface is CQRS
+
+CQRS (Command and Query Responsibility Segregation) is a pattern that separates read and write operations for a data store solution. In our case the credit and debit events are written into a database different from the database which the data is read from.
```

Comment:
C'est pas hyper facile de comprendre la dernière phrase `In our case the credit and debit events are written into a database different from the database which the data is read from.`

Reply:
Le CQRS n'est pas un pattern de stockage de data, ça n'a pas de lien avec le data store. C'est un pattern de séparation des opérations sur l'interface (HTTP par exemple) d'un système.

Reply:
Le CQRS n'est pas un pattern de stockage de data, ça n'a pas de lien avec le data store. C'est un pattern de séparation des opérations sur l'interface (HTTP par exemple) d'un système.